File: DSA/Ayush_Mishra_590014844/Day_17_Question_2.c
#include <stdio.h>

int firstOccurrence(int arr[], int size, int target) {
    int left = 0, right = size - 1, result = -1;
    while (left <= right) {
        int mid = left + (right - left) / 2;
        if (arr[mid] == target) {
            result = mid;
            right = mid - 1;
        } else if (arr[mid] < target) left = mid + 1;
        else right = mid - 1;
    }
    return result;
}

int main() {
    int n, target;
    printf("Enter number of elements: ");
    scanf("%d", &n);
    int arr[n];
    printf("Enter sorted elements:\n");
    for (int i = 0; i < n; i++) scanf("%d", &arr[i]);
    printf("Enter target element: ");
    scanf("%d", &target);
    int index = firstOccurrence(arr, n, target);
    printf("First occurrence index: %d\n", index);
    return 0;
}

File: DSA/Ayush_Mishra_590014844/Day_17_Question_2.py
def first_occurrence(arr, target):
    left, right = 0, len(arr) - 1
    result = -1
    while left <= right:
        mid = (left + right) // 2
        if arr[mid] == target:
            result = mid
            right = mid - 1
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1
    return result

arr = list(map(int, input("Enter sorted array with duplicates: ").split()))
target = int(input("Enter target element: "))
print("First occurrence index:", first_occurrence(arr, target))
